app/ui/src/app/integration/detail-page/detail.component.ts

```diff
@@ -145,19 +160,18 @@ export class IntegrationDetailComponent implements OnInit, OnDestroy {
         // TODO doesn't this just mean edit?
         break;
       case STOP_INTEGRATION:
-        /*
-        this.integrationActionsService.requestAction('deactivate', this.integration);
-        */
+        this.integrationActionsService.requestAction('deactivate', <any> this.integration);
         break;
       case PUBLISH:
-        /*
         {
-          const integration = { ...this.integration };
-          integration.steps = deployment.spec.steps;
-          //this.integration.desiredStatus = PUBLISHED;
-          this.integrationActionsService.requestAction('publish', integration);
+          const integration = {
+            integrationVersion: this.integration.version,
+            id: this.integration.id,
+            name: this.integration.name,
+            version: deployment.version
+           };
+          this.integrationActionsService.requestAction('publish', <any> integration);
```

Comment:
@chirino I doubt I'm doing this right, this results in a `PUT` to `/integrations/{id}/deployments/{version}` with `{ targetState: 'Published' }` but that results in an HTTP 405, I thought that was how this was supposed to work?